### PR TITLE
Add recursive type contains function

### PR DESCRIPTION
**Pull request by bryancroteau-MSFT**
To be used by the JSON function in PowerApps.

### CODE REVIEW

**Comment by BruceHaley** [comment]
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Connectors.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Core.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Interpreter.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.LanguageServerProtocol.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Transport.Attributes.dll**

File: src/libraries/Microsoft.PowerFx.Core/Types/DType.cs [hunk 1]
```diff
@@ -1564,6 +1564,75 @@ public DType DropMulti(ref bool fError, DPath path, params DName[] rgname)
...
+
+        public bool ContainsKindNested(DPath path, DKind kind)
+        {
+            AssertValid();
+            Contracts.Assert(kind >= DKind._Min && kind < DKind._Lim);
+
+            var fullType = this;
+            if (IsLazyType)
+            {
+                // This probably should throw (or be eliminated)
```
Comment (lesaltzm) [inline src/libraries/Microsoft.PowerFx.Core/Types/DType.cs:1604]: Yeah. For Lazy Types, all these *Nested operations should only look at the top layer and not recurse, imo.

Comment (lesaltzm) [reply src/libraries/Microsoft.PowerFx.Core/Types/DType.cs:1604]: We can then have runtime errors if something deeper than that violates the constraint we tried to enforce with this type function.

Comment (bryancroteau-MSFT) [reply src/libraries/Microsoft.PowerFx.Core/Types/DType.cs]: Done

**Comment by BruceHaley** [comment]
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Connectors.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Core.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Interpreter.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.LanguageServerProtocol.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Transport.Attributes.dll**

**Review by lesaltzm — APPROVED** [review]

**Pull request merged**